time-series/time-series-api/src/main/java/com/powsybl/timeseries/ast/BinaryMaxCalc.java

```diff
@@ -52,6 +53,6 @@ public boolean equals(Object obj) {
 
     @Override
     public int hashCode() {
-        return left.hashCode() + right.hashCode();
+        return Objects.hash(left, right, NAME);
```

Comment:
Why do you need to add the NAME to the hash calculation?

Reply:
If I don't, a BinaryMinCalc and a BinaryMaxCalc with the same child nodes would have the same hashcode